index.ts

```diff
@@ -19,7 +19,10 @@ async function run(): Promise<void> {
     let deployBranch = core.getInput('deploy-branch')
     if (!deployBranch) deployBranch = DEFAULT_DEPLOY_BRANCH
 
-    if (github.context.ref === `refs/heads/${deployBranch}`) {
+    const deployRepo = core.getInput('deploy-repo')
+    const isSameRepo = !deployRepo || deployRepo === github.context.repo.repo
+
+    if (isSameRepo && github.context.ref === `refs/heads/${deployBranch}`) {
```

Comment:
Would you mind adding a new test to validate this? 😃 

There is an existing one that checks skipping the build if the branch is the same: https://github.com/enriikke/gatsby-gh-pages-action/blob/84aa969a7918a9488d4dac31f2d847274b31ff07/__tests__/index.test.ts#L72-L77

I'm thinking we need a similar one that actually calls `build` if the branches are the same but the repo is different. I think you could validate the `build` call at the end by doing something like this: https://github.com/enriikke/gatsby-gh-pages-action/blob/84aa969a7918a9488d4dac31f2d847274b31ff07/__tests__/index.test.ts#L84

Reply:
New test case is added, see new comments for further info.